lib/SPIRV/SPIRVBuiltinHelper.cpp

```diff
@@ -64,9 +64,15 @@ BuiltinCallMutator::BuiltinCallMutator(
     : CI(CI), FuncName(FuncName),
       Attrs(CI->getCalledFunction()->getAttributes()), ReturnTy(CI->getType()),
       Args(CI->args()), Rules(Rules), Builder(CI) {
-  getParameterTypes(CI->getCalledFunction(), PointerTypes,
-                    std::move(NameMapFn));
-  PointerTypes.resize(Args.size(), nullptr);
+  bool DidDemangle = getParameterTypes(CI->getCalledFunction(), PointerTypes,
+                                       std::move(NameMapFn));
+  if (!DidDemangle) {
+    // TODO: PipeBlocking.ll causes demangling failures.
+    // assert(isNonMangledOCLBuiltin(CI->getCalledFunction()->getName()) &&
```

Comment:
It does look like that if a function name has `.1` in the end (for example 
`declare spir_func void @_Z30__spirv_WritePipeBlockingINTELPU3AS115__spirv_Pipe__1PU3AS4iii.1(%opencl.pipe_wo_t addrspace(1)* %0, i9 addrspace(4)* %1, i32 %2, i32 %3)`
RootNode will be `nullptr`
`auto *RootNode = dyn_cast_or_null<FunctionEncoding>(Demangler.parse());`
unsure if it's expected

Reply:
If the `.1` is added, then the function name isn't a valid mangled name which is... probably problematic, but that requires some more investigation.

Reply:
Agree